src/youtube.js

```diff
@@ -0,0 +1,1104 @@
+   * 3 = HAVE_FUTURE_DATA - data for the current and at least the next frame is available.
+   * 4 = HAVE_ENOUGH_DATA - enough data available to start playing.
+   */
+  get readyState(): number {
+    if (!this._api) {
+      return 0;
+    }
+    const playerState = window.YT.PlayerState;
+    switch (this._api.getPlayerState()) {
+      case playerState.UNSTARTED:
+        return 0;
+      case playerState.BUFFERING:
+        return 2;
+      default:
+        return 4;
+    }
+  }
+
+  /**
+   * @returns {Number} - The height of the video player, in pixels.
+   * @public
+   */
+  get videoHeight(): number {
+    return -1;
+  }
+
+  /**
+   * @returns {Number} - The width of the video player, in pixels.
+   * @public
+   */
+  get videoWidth(): number {
+    return -1;
+  }
+
+  /**
+   * @param {boolean} playsinline - Whether to set on the video tag the playsinline attribute.
+   */
+  set playsinline(playsinline: boolean): void {}
+
+  /**
+   * @returns {boolean} - Whether the video tag has an attribute of playsinline.
+   */
+  get playsinline(): boolean {
+    return this._config.playsinline;
+  }
+
+  /**
+   * Set crossOrigin attribute.
+   * @param {?string} crossOrigin - 'anonymous' or 'use-credentials'
+   */
+  set crossOrigin(crossOrigin: ?string): void {}
+
+  /**
+   * Get crossOrigin attribute.
+   * @returns {?string} - 'anonymous' or 'use-credentials'
+   */
+  get crossOrigin(): ?string {
+    return null;
+  }
+
+  /**
+   * get the playback rates
+   * @return {number[]} - playback rates
+   */
+  get playbackRates(): Array<number> {
+    return this._api ? this._api.getAvailablePlaybackRates() : Youtube.PLAYBACK_RATES;
+  }
+
+  /**
+   * Initializes the engine.
+   * @param {PKMediaSourceObject} source - The selected source object.
+   * @param {Object} config - The player configuration.
+   * @private
+   * @returns {void}
+   */
+  _init(source: PKMediaSourceObject, config: Object): void {
+    this._source = source;
+    this._config = config;
+    const dispatchError = e => {
+      const error = new Error(Error.Severity.CRITICAL, Error.Category.PLAYER, Error.Code.LOAD_FAILED, e);
+      this.dispatchEvent(new FakeEvent(EventType.ERROR, error));
+    };
+    this._sdkLoaded = new Promise((resolve, reject) => {
+      this._apiReady = () => {
+        if (this._config.playback.muted) {
+          this.muted = true;
+        }
+        resolve();
+      };
+      this._apiError = (e) => {
+        const error = this._getYoutubeErrorCodeData(e.data);
+        dispatchError(error);
+        reject(e.data);
+      };
+    });
+    this._loadYouTubeIframeAPI().then(() => {
+      this._loadYouTubePlayer();
+    }).catch(e => {
+      dispatchError(e);
+    });
+  }
+
+  _getYoutubeErrorCodeData(errorNumber: number): {code: number, message: string} {
+    switch (errorNumber) {
+      case 2:
```

Comment:
not sure if we should have enum for the error numbers from youtube as well

Reply:
they are not enumed by Youtube so didn't add them as our enums as well